sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -301,6 +302,16 @@ object CatalystTypeConverters {
       DateTimeUtils.toJavaTimestamp(row.getLong(column))
   }
 
+  private object IntervalConverter extends CatalystTypeConverter[Interval, Interval, Array[Byte]] {
+    override def toCatalystImpl(scalaValue: Interval): Array[Byte] =
```

Comment:
@adrian-wang also suggested using a single long to encode -- we should consider and decide on that as well.